exporter/terra/gcs.py

```diff
@@ -95,15 +105,17 @@ def transfer_upload_area(self, source_bucket: str, upload_area_key: str, export_
             else:
                 raise
 
-    def transfer_job_for_upload_area(self, source_bucket: str, upload_area_key: str, export_job_id: str) -> TransferJobSpec:
+    def transfer_job_for_upload_area(self, source_bucket: str, upload_area_key: str, project_uuid: str, export_job_id: str) -> TransferJobSpec:
         return TransferJobSpec(name=f'transferJobs/{export_job_id}',
                                description=f'Transfer job for ingest upload-service area {upload_area_key} and export-job-id {export_job_id}',
                                project_id=self.project_id,
                                source_bucket=source_bucket,
-                               source_key=upload_area_key,
-                               dest_bucket=self.gcs_dest_bucket,
+                               source_path=f'{upload_area_key}/',
                                aws_access_key_id=self.aws_access_key_id,
-                               aws_access_key_secret=self.aws_access_key_secret)
+                               aws_access_key_secret=self.aws_access_key_secret,
+                               dest_bucket=self.gcs_dest_bucket,
+                               dest_path=f'{self.gcs_bucket_prefix}/{project_uuid}/data/',
```

Comment:
Kudos on coordinating this spec change with other DCP2 devs! It really made things simple. 

We do have an assumption now that the all data files will be in the staging area and this may change in 6mos when they cleaned it up so we may need to do some sort of tracking which things has already been imported.

Reply:
if that happens then basically the exporter will export everything to the terra staging area again incurring the time and transfer cost. we currently do not have a way for selective export, but that was one of the thing i played around with and can be achieved in a number of ways for e.g. checkbox in ui to select/unselect data export for updates, or implicitly check in the implementation itself instead of relying on the transfer service overwrite rule.